openvdb_points/tools/PointGroup.h

```diff
@@ -0,0 +1,588 @@
+
+    typedef typename tree::LeafManager<PointDataTreeType>       LeafManagerT;
+    typedef typename LeafManagerT::LeafRange                    LeafRangeT;
+    typedef AttributeSet::Descriptor::NameAndType               NameAndType;
+    typedef AttributeSet::Descriptor::GroupIndex                GroupIndex;
+
+    CopyGroupOp(PointDataTreeType& tree,
+                const GroupIndex& targetIndex,
+                const GroupIndex& sourceIndex)
+        : mTree(tree)
+        , mTargetIndex(targetIndex)
+        , mSourceIndex(sourceIndex) { }
+
+    void operator()(const typename LeafManagerT::LeafRange& range) const {
+
+        for (typename LeafManagerT::LeafRange::Iterator leaf=range.begin(); leaf; ++leaf) {
+
+            GroupHandle sourceGroup = leaf->groupHandle(mSourceIndex);
+            GroupWriteHandle targetGroup = leaf->groupWriteHandle(mTargetIndex);
+
+            for (IndexIter iter = leaf->beginIndexAll(); iter; ++iter) {
+                const bool groupOn = sourceGroup.get(*iter);
+                targetGroup.set(*iter, groupOn);
+            }
+        }
+    }
+
+    //////////
+
+    PointDataTreeType&      mTree;
+    const GroupIndex        mTargetIndex;
+    const GroupIndex        mSourceIndex;
+};
+
+
+template <typename PointDataTree, typename PointIndexTree>
+struct SetGroupFromIndexOp
+{
+    typedef typename tree::LeafManager<PointDataTree>   LeafManagerT;
+    typedef typename LeafManagerT::LeafRange            LeafRangeT;
+    typedef typename PointIndexTree::LeafNodeType       PointIndexLeafNode;
+    typedef typename PointIndexLeafNode::IndexArray     IndexArray;
+    typedef AttributeSet::Descriptor::GroupIndex        GroupIndex;
+    typedef std::vector<bool>                           BoolArray;
+
+    SetGroupFromIndexOp(const PointIndexTree& indexTree,
+                        const BoolArray& membership,
+                        const GroupIndex& index)
+        : mIndexTree(indexTree)
+        , mMembership(membership)
+        , mIndex(index) { }
+
+    void operator()(const typename LeafManagerT::LeafRange& range) const
+    {
+        for (typename LeafManagerT::LeafRange::Iterator leaf=range.begin(); leaf; ++leaf) {
+
+            // obtain the PointIndexLeafNode (using the origin of the current leaf)
+
+            const PointIndexLeafNode* pointIndexLeaf = mIndexTree.probeConstLeaf(leaf->origin());
+
+            if (!pointIndexLeaf)    continue;
+
+            // obtain the group attribute array
+
+            GroupWriteHandle group(leaf->groupWriteHandle(mIndex));
+
+            // initialise the attribute storage
+
+            pointIndexLeaf->indices();
+
+            Index64 index = 0;
+
+            const IndexArray& indices = pointIndexLeaf->indices();
+
+            for (typename IndexArray::const_iterator it = indices.begin(), it_end = indices.end(); it != it_end; ++it)
+            {
+                if (mMembership.at(*it)) {
+                    group.set(index, true);
```

Comment:
The assumption was in the typical case you'd be using this from a conversion so the group would be empty at the start and you'd be doing inclusive membership. I realise that might not necessarily be the case, so I've added a new parameter with a default value so that you can optionally disable this functionality and use it like you've specified if you wish.